services/core-web/src/components/mine/Permit/PermitConditions.tsx

```diff
@@ -65,14 +74,43 @@ const PermitConditions: FC<PermitConditionProps> = ({
         permitConditions?.filter(
           (c) => c.condition_category_code === cat.condition_category_code
         ) ?? [];
+
+      // Recursive function to get the full path of steps
+      const getStepPath = (condition, parentPath = ""): IPermitCondition => {
+        const currentPath = parentPath
+          ? `${parentPath}${condition.step}`
+          : `${cat.description} - ${condition.step}`;
+        const stepPath = currentPath.replace(/\.+$/, "");
+
+        const mineReportPermitRequirement = mineReportPermitRequirements.find(
+          (requirement) =>
+            requirement.permit_condition_id === condition.permit_condition_id
+        );
+
+        // If condition has sub-conditions, recursively add step paths
+        const sub_conditions =
+          condition.sub_conditions?.map((subCondition) => getStepPath(subCondition, currentPath)) ??
+          [];
+
+        return {
+          ...condition,
+          stepPath,
+          mineReportPermitRequirement,
+          sub_conditions,
+        };
+      };
+
+      // Initialize the step paths for all top-level conditions
+      const updatedConditions = conditions.map((condition) => getStepPath(condition));
```

Comment:
perhaps a bit of a nitpick, but are these _updated_ conditions, as in they have updates that the user is making that need to be saved? Or are they _formatted_?

Reply:
Fair enough.  I'll update it!